src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -66,4 +80,105 @@ public String version() {
     public ConfigDef config() {
         return SplunkSinkConnectorConfig.conf();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues =
+            config.configValues()
+                .stream()
+                .collect(Collectors.toMap(
+                    ConfigValue::name,
+                    Function.identity()));
+
+        SplunkSinkConnectorConfig connectorConfig;
+        try {
+            connectorConfig = new SplunkSinkConnectorConfig(connectorConfigs);
+        } catch (Exception e) {
+            log.error("Validating configuration caught an exception", e);
+            return config;
+        }
+
+        validateAccess(connectorConfig, configValues);
```

Comment:
We might still be able to decouple the access and invalid hostname separately by using `InetAddress.getByName(...)`. 


In fact, we can also add checks for reachability by simply creating a socket
```java
Socket s = new Socket();
s.connect(new InetSocketAddress(host, port), 100);
```
and then proceed to check accesses.

Reply:
And what happens right now is that if it is unable to connect, the exception is thrown that connection failed against the specific host. If it does go through the full run, it would succeed instead.